Add metadata plugin

JakobMiksch opened the pull request:
resolves #240 

It shows the "Metadata" option for all layers, like specified in the description. 

However, metadata can only be retrieved for WMS and WFS layers from MapServer. 

Metadata for Vectortiles could also be possible, but the layername and the baseurl are difficult to extract, since they are not separetely added in the config file (like for WMS and WFS). Both layername and baseurl could of course be extracted from the connection-url:
`https://w08-mapserver.compass.ie/mapserver/?map=/MapServer/apps/mapview-demo/example.map&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2Fpng&TRANSPARENT=true&LAYERS=waterways&TILED=false&CRS=EPSG%3A3857&WIDTH={width}&HEIGHT={height}&BBOX={bbox}&STYLES=Waterways&FORMAT=mvt`
However, it might be more elegant and consistent to provide the layername and the baseurl directly in the config file. 

If no metadata is available, an alert window is shown. This is probably not the behavior you would like for long term.
@geographika what is your preferred way to display "no metadata"?

chrismayer commented on app/view/menuitem/LayerMetadata.js:
```diff
@@ -0,0 +1,173 @@
+         * Helper function to notify user
+         * that metadata is not availble
+         */
+        function alert_no_metadata(){
```
Please move this outside of handlerFunc as member function. Also please use camel case notation

JakobMiksch replied:
fixed with 8e7108b5c8286adfce0795e7c372108b3748adf2

chrismayer reviewed (approved):
LGTM thanks, I added a minor thing. Once @geographika gave his feedback this ist good to go IMHO after you made Travis happy ;)

JakobMiksch commented:
this commit 5755e42eeb6ae3787d7aee17f8e2028ef79f098d makes ESLint and therefore Travis and Azure happy

geographika commented on app/view/menuitem/LayerMetadata.js:
```diff
@@ -0,0 +1,171 @@
+                    }
+                    // metadata is fine and can be displayed
+                    else{
+                        var windowTitle = 'Metadata '+ me.layer.get('name');
```
Would read better as '{layername} Metadata'

geographika commented:
@JakobMiksch - works great - thanks!

A few comments:

- I presume the ISO schemas can be downloaded and hosted for the project if required without any issues?
- To handle missing metadata I'd suggest the Metadata context-menuitem is only shown for a layer with a `hasMetadata: true` flag in the config. This would be easy to set for all layers of a certain type such as WMS/WFS using the type defaults here: https://github.com/compassinformatics/cpsi-mapview/blob/2410393dadaadc783eb51e3e1fb7006a093e0297/resources/data/layers/default.json#L22
- If MD is still missing an ExtJS message box with a warning should be displayed
- For vector tiles setting `layername` and the `baseurl` directly in the config file sounds a good approach. It would just require setting these properties in the `default.json`?

JakobMiksch commented:
@geographika I addressed your remarks:

- yes, the ISO schema can also be downloaded - maybe @chrismayer can elaborate further if necessary
- I added the flag `hasMetadata: true`
- the alert for missing metadata is replaced with a Ext message box
- the vector tile config now has the flags `baseUrl` and `layerIdentificationName`. The rather long name `layerIdentificationName` was intentionally chosen in order to avoid confusions with flag `name` which is actually the title
- I added `CpsiMapview.view.menuitem.LayerMetadata` to the "requires" of the layer tree component

geographika commented:
Thanks @JakobMiksch !
